mkosi/__init__.py

```diff
@@ -3466,12 +3456,19 @@ def invoke_repart(
 
     output = json.loads(run(cmdline, stdout=subprocess.PIPE).stdout)
 
+    roothash = usrhash = None
     for p in output:
-        if p["type"].startswith("usr") or p["type"].startswith("root"):
-            usr_only = p["type"].startswith("usr")
-            return (p["label"], p.get("roothash"), usr_only)
+        if (h := p.get("roothash")) is None:
+            continue
+
+        assert p["type"].startswith("usr") or p["type"].startswith("root")
+
+        if p["type"].startswith("usr"):
+            usrhash = usrhash or h
+        else:
+            roothash = roothash or h
```

Comment:
So, the first one of either we find always wins, e.g. having multiple `/usr/` in a single image is nothing we need to think about? Either way a comment would be helpful for the future.